### PR TITLE
Contact Methods reorg

### PR DESCRIPTION
**Pull request by Loquacity**
# Description

Reorganises the client contact methods section.

TO COME IN A DIFFERENT PR: 
* Reorg registration section
* Add new contact method section for Salt Minion

# Target branches

Which product version does this PR apply to?

- [ ] Manager-4.0
- [ ] Manager-4.1
- [x] Manager-4.2
- [x] Uyuni

# Links

Partial https://github.com/SUSE/spacewalk/issues/12477

### CODE REVIEW

**Comment by Loquacity** [comment]
image

**Review by keichwa — APPROVED** [review]

**Review by jcayouette — APPROVED** [review]
Nice work looks good!

File: modules/client-configuration/nav-client-configuration-guide.adoc [hunk 1]
```diff
@@ -27,11 +27,14 @@ endif::[]
 ** xref:gpg-keys.adoc[GPG Keys]
 ** xref:activation-keys.adoc[Activation Keys]
 ** xref:bootstrap-repository.adoc[Bootstrap Repository]
+// Contact Methods
 ** xref:contact-methods-intro.adoc[Client Contact Methods]
```
Comment (paususe) [inline modules/client-configuration/nav-client-configuration-guide.adoc:31]:
```suggestion
** xref:contact-methods-intro.adoc[Client Management Methods]
```

File: modules/client-configuration/nav-client-configuration-guide.adoc [hunk 2]
```diff
@@ -27,11 +27,14 @@ endif::[]
...
-*** xref:contact-methods-rhnsd.adoc[rhnsd]
-*** xref:contact-methods-pushssh.adoc[Push via SSH]
-*** xref:contact-methods-saltssh.adoc[Push via Salt SSH]
-*** xref:contact-methods-osad.adoc[OSAD]
+*** xref:contact-methods-salt.adoc[Contact Methods for Salt Clients]
```
Comment (paususe) [inline modules/client-configuration/nav-client-configuration-guide.adoc:32]:
```suggestion
*** xref:contact-methods-salt.adoc[Management as Salt Clients]
```

File: modules/client-configuration/nav-client-configuration-guide.adoc [hunk 3]
```diff
@@ -27,11 +27,14 @@ endif::[]
...
+**** xref:contact-methods-saltssh.adoc[Push via Salt SSH]
+*** xref:contact-methods-traditional.adoc[Contact Methods for Traditional Clients]
```
Comment (paususe) [inline modules/client-configuration/nav-client-configuration-guide.adoc:34]:
```suggestion
*** xref:contact-methods-traditional.adoc[Management as Traditional Clients]
```

File: modules/client-configuration/nav-client-configuration-guide.adoc [hunk 4]
```diff
@@ -27,11 +27,14 @@ endif::[]
...
+**** xref:contact-methods-rhnsd.adoc[rhnsd]
+**** xref:contact-methods-pushssh.adoc[Push via SSH]
+**** xref:contact-methods-osad.adoc[OSAD]
```
Comment (paususe) [inline modules/client-configuration/nav-client-configuration-guide.adoc:37]:
```suggestion
**** xref:contact-methods-rhnsd.adoc[rhnsd]
***** xref:contact-methods-osad.adoc[OSAD]
**** xref:contact-methods-pushssh.adoc[Push via SSH]
```

File: modules/client-configuration/pages/contact-methods-traditional.adoc [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+Traditional clients can communicate with the {productname} Server using a range of methods.
+
+The {productname} daemon ([command]``rhnsd``) runs on traditional client systems and periodically connects with {productname} to check for new updates and notifications.
+
+Push via SSH is used in environments where clients cannot reach the {productname} Server directly.
+In this environment, clients are located in a firewall-protected zone called a DMZ.
+No system within the DMZ is authorized to open a connection to the internal network, including the {productname} Server.
+
+OSAD is an alternative contact method between {productname} and its clients.
+OSAD allows registered client systems to execute scheduled actions immediately.
```
Comment (paususe) [inline modules/client-configuration/pages/contact-methods-traditional.adoc:13]:
```suggestion
OSAD is an alternative contact method between {productname} and its clients which allows registered client systems to execute scheduled actions immediately. OSAD is typically used conjunction with rhnsd.
```

**Review by paususe — CHANGES_REQUESTED** [review]
Some suggestions.

Also, we are missing a page for Salt Minions under "Management as Salt Clients".

File: modules/client-configuration/nav-client-configuration-guide.adoc (hunk 2 above)
Comment (Loquacity) [reply modules/client-configuration/nav-client-configuration-guide.adoc:32]: The preposition doesn't make sense here, it should be "of". How about starting with the verb instead: "Manage Salt Clients"?

**Comment by Loquacity** [comment]
> Some suggestions.
> 
> Also, we are missing a page for Salt Minions under "Management as Salt Clients".

Yep, that'll be in a different PR. Trying to make sure I chunk this out properly.

**Comment by keichwa** [comment]
Lana Brindley <notifications@github.com> writes:

> The preposition doesn't make sense here, it should be "of". How about
> starting with the verb instead: "Manage Salt Clients"?

I think this comes closer.  But actually we want to talk about some ways
to manage clients.  I'd propose sth like this:

"Use Salt to Manage Clients"

And then we have different methods to make use of Salt.  The same with
the traditional stack.

Or vice versa ;)

-- 
Karl Eichwalder

**Comment by Loquacity** [comment]
> I think this comes closer. But actually we want to talk about some ways to manage clients. I'd propose sth like this: "Use Salt to Manage Clients" And then we have different methods to make use of Salt. The same with the traditional stack. Or vice versa ;)

It's not really "using Salt" though, it's "using <various contact methods> to talk to Salt clients". So the "salt" is modifying the client, not the subject of the sentence, even though it feels like the subject because we're using it to define the chapter. Also, what's the traditional version? You can't have "Use traditional to manage clients" :wink:

**Review by keichwa — APPROVED** [review]

**Pull request merged**